pbj-core/pbj-runtime/src/test/java/com/hedera/pbj/runtime/ProtoWriterToolsTest.java

```diff
@@ -0,0 +1,759 @@
+import static com.hedera.pbj.runtime.ProtoWriterTools.TAG_TYPE_BITS;
+import static com.hedera.pbj.runtime.ProtoWriterTools.sizeOfBooleanList;
+import static com.hedera.pbj.runtime.ProtoWriterTools.sizeOfBytesList;
+import static com.hedera.pbj.runtime.ProtoWriterTools.sizeOfDelimited;
+import static com.hedera.pbj.runtime.ProtoWriterTools.sizeOfDoubleList;
+import static com.hedera.pbj.runtime.ProtoWriterTools.sizeOfEnumList;
+import static com.hedera.pbj.runtime.ProtoWriterTools.sizeOfFloatList;
+import static com.hedera.pbj.runtime.ProtoWriterTools.sizeOfIntegerList;
+import static com.hedera.pbj.runtime.ProtoWriterTools.sizeOfLongList;
+import static com.hedera.pbj.runtime.ProtoWriterTools.sizeOfMessageList;
+import static com.hedera.pbj.runtime.ProtoWriterTools.sizeOfStringList;
+import static com.hedera.pbj.runtime.ProtoWriterTools.wireType;
+import static com.hedera.pbj.runtime.ProtoWriterTools.writeBytes;
+import static com.hedera.pbj.runtime.ProtoWriterTools.writeInteger;
+import static com.hedera.pbj.runtime.ProtoWriterTools.writeLong;
+import static com.hedera.pbj.runtime.ProtoWriterTools.writeMessage;
+import static com.hedera.pbj.runtime.ProtoWriterTools.writeOptionalBoolean;
+import static com.hedera.pbj.runtime.ProtoWriterTools.writeOptionalBytes;
+import static com.hedera.pbj.runtime.ProtoWriterTools.writeOptionalDouble;
+import static com.hedera.pbj.runtime.ProtoWriterTools.writeOptionalFloat;
+import static com.hedera.pbj.runtime.ProtoWriterTools.writeOptionalInteger;
+import static com.hedera.pbj.runtime.ProtoWriterTools.writeOptionalLong;
+import static com.hedera.pbj.runtime.ProtoWriterTools.writeOptionalString;
+import static com.hedera.pbj.runtime.ProtoWriterTools.writeString;
+import static com.hedera.pbj.runtime.ProtoWriterTools.writeTag;
+import static java.util.Arrays.asList;
+import static java.util.Collections.emptyList;
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.hedera.pbj.runtime.io.buffer.BufferedData;
+import com.hedera.pbj.runtime.io.buffer.Bytes;
+import net.bytebuddy.utility.RandomString;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.EnumSource;
+import test.proto.Apple;
+
+
+import java.io.IOException;
+import java.nio.ByteOrder;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.random.RandomGenerator;
+
+class ProtoWriterToolsTest {
+
+    public static final int SIZE_OF_TAG = 1;
+    public static final int SIZE_OF_LENGTH_VAR = 1;
+    public static final int MAX_VAR_INT_SIZE = 6;
+    public static final int MAX_VAR_FLOAT_SIZE = 5;
```

Comment:
What does this mean, exactly?

My math is: max value here is when field ID is max (like 2^29 - 1), in this case total size of non-null optional float will be 5 (delimited tag) + 5 (delimited length) + 4 (fixed float)

Reply:
Yes, it's confusing. Renamed the constant and used a sum of `FLOAT_SIZE` and `TAG_SIZE`